### DISCUSSION TITLE
SQLSTATE[HY000] [1698] Access denied for user 'root'@'localhost' error on install

### DISCUSSION THREAD

**Discussion by MabelleLuna** [pimcore/pimcore#16018] [Q&A / Support] [answered]
Hello, I ran into an issue when trying to install pimcore in Ubuntu via docker with this command:

`docker run -u `id -u`:`id -g` --rm -v `pwd`:/var/www/html pimcore/pimcore:php8.2-latest composer create-project pimcore/demo pim`

After some time, it threw this error and timed out:

```
In Process.php line 1204:

  The process "'/usr/local/bin/php' 'bin/console' 'cache:clear' '--no-warmup'
   '--ignore-maintenance-mode'" exceeded the timeout of 900 seconds.
```

When I go to my pim public folder it shows this error:
`An exception occurred in the driver: SQLSTATE[HY000] [1698] Access denied for user 'root'@'localhost'`

I can't find any resolutions on this other than this: https://talk.pimcore.org/t/switch-to-production-mode-in-5-4-causes-fatal-error-access-denied/1234/3

I realized my database wasn't set up prior to installation so I did set that up along with a user granted all privelages but now what? Is there a file I need to update to add my db credentials so the pimcore install can finish?

Also is there a way I can rerun the pimcore installation or continue it or would I need to remove pimcore and reinstall it again with the docker command?

Any help is welcome, thanks!

**Comment by dvesh3** [comment]
Hi @MabelleLuna you can run installer again with `docker compose exec php vendor/bin/pimcore-install --mysql-host-socket=db --mysql-username=pimcore --mysql-password=pimcore --mysql-database=pimcore` and passing correct db credentials. This will re-install it with fresh data.

**Reply by MabelleLuna** [reply]
Thanks @dvesh3 I tried that command within the pim directory and it says service "php" not running. I have php files that are executed when I test. Is there a reason for this or is there another module I need in order to run it within the pim directory? I tried looking this up but did not find anything on this. Thanks in advance!

**Reply by dvesh3** [reply]
you need to follow 2 & 3 steps first https://github.com/pimcore/skeleton#follow-these-steps

**Reply by MabelleLuna** [reply]
@dvesh3 it looks like that is the Skeleton version. I am installing the Demo, that should not matter correct?

**Reply by MabelleLuna** [reply, marked as answer]
Update: We ended up switching to a premade VM and that seemed to work because installing pimcore worked right away. There might have been missing libraries/packages on our previous VPS because the demo install wouldn't work at all.
Anyway thanks for helping!
